### ISSUE TITLE
pass-server-data example: Use API Routes?

### ISSUE THREAD

**Issue by karlhorky** [zeit/next.js#8976]
# Examples bug report

## Example name

[`pass-server-data`](https://github.com/zeit/next.js/tree/0cd2732c7f511dc28da62c8ac9b38bf9522ca960/examples/pass-server-data)

## Describe the bug

So not as much of a bug as a suggestion of a potential improvement:

The data is currently being `require`d and passed to the pages by [a custom `server.js` file](https://github.com/zeit/next.js/blob/0cd2732c7f511dc28da62c8ac9b38bf9522ca960/examples/pass-server-data/server.js) that uses Express.

Would the Next.js team be open to migrating this example to use [API Routes](https://nextjs.org/blog/next-9#api-routes)? Maybe with an implementation along the lines of [React Tricks: Exploring Next.js 9 Dynamic Routing & API Routes](https://reacttricks.com/exploring-next-9-dynamic-routing-and-api-routes/)?

## Alternatives

A potential alternative (which would allow retaining the current version of `pass-server-data` as it is for reference) would be to copy and rename this example to create a new example that did use API Routes.

## To Reproduce

Not applicable

## Expected behavior

Example uses API Routes

## System information

Not Applicable

## Additional context

-

**Comment by timneutkens** [comment]
I think the example should be removed, it shows a bunch of bad practices.

**Comment by karlhorky** [comment]
Ah, ok! Can you list these bad practices out?

Maybe these two tasks could be done at the same time? Eg:

1. Convert the example to use API Routes
2. Refactor the example to remove bad practices

**Comment by timneutkens** [comment]
The goal of the example is showing passing data as the querystring which in general you don't want to do, `getInitialProps` covers fetching data.

**Comment by karlhorky** [comment]
Ok, so if I was to rework the example in the way described in [the blog post](https://reacttricks.com/exploring-next-9-dynamic-routing-and-api-routes/), this would be acceptable?

TLDR version:

- Set up an API route page that `require`s the data
- `isomorphic-unfetch` from that API route in `getInitialProps`

**Comment by timneutkens** [comment]
I guess that might work, but that's already covered here: https://github.com/zeit/next.js/tree/canary/examples/api-routes

**Comment by karlhorky** [comment]
Ok great! Don't need to do anything except remove the `pass-server-data` example then!

Should I open a PR to do this?

**Comment by timneutkens** [comment]
Sure!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
